tests/test_frame.py

```diff
@@ -1145,20 +1145,39 @@ def test_types_regressions() -> None:
 
 
 def test_read_csv() -> None:
-    if TYPE_CHECKING:  # skip pytest
-        #  https://github.com/microsoft/python-type-stubs/issues/87
-        df11: pd.DataFrame = pd.read_csv("foo")
-        df12: pd.DataFrame = pd.read_csv("foo", iterator=False)
-        df13: pd.DataFrame = pd.read_csv("foo", iterator=False, chunksize=None)
-        df14: TextFileReader = pd.read_csv("foo", chunksize=0)
-        df15: TextFileReader = pd.read_csv("foo", iterator=False, chunksize=0)
-        df16: TextFileReader = pd.read_csv("foo", iterator=True)
-        df17: TextFileReader = pd.read_csv("foo", iterator=True, chunksize=None)
-        df18: TextFileReader = pd.read_csv("foo", iterator=True, chunksize=0)
-        df19: TextFileReader = pd.read_csv("foo", chunksize=0)
+    with ensure_clean() as path:
+        Path(path).write_text("A,B\n1,2")
+        check(assert_type(pd.read_csv(path), pd.DataFrame), pd.DataFrame)
+        check(
+            assert_type(pd.read_csv(path, iterator=False), pd.DataFrame), pd.DataFrame
+        )
+        check(
+            assert_type(
+                pd.read_csv(path, iterator=False, chunksize=None), pd.DataFrame
+            ),
+            pd.DataFrame,
+        )
+
+        with assert_type(pd.read_csv(path, chunksize=1), TextFileReader):
```

Comment:
`DataFrame` has no `__exit__`/`__enter__` so it would fail anyways. I think the simplest would be to make `check` return the object to mirror `assert_type`. Then we could do

```py
with check(assert_type(pd.read_csv(path, chunksize=1), TextFileReader), TextFileReader):
    pass
```

Fine with either `...` or `pass`.

Reply:
need to call `__exit__` to close the file handle, otherwise `ensure_clean` cannot remove the file on windows.